### PR TITLE
Panel Edit: Fix data links edit icons being off screen when provided title is too long

### LINKED ISSUE #59150: Data-links edit and delete icons are off-screen and not visible
**What happened**:

**See this video for reproduction steps:
https://www.loom.com/share/e4874e7f4e884258a43d5f8e7e185f24**

Simpler reproduction steps:
1. Find a table panel
2. Add an override for one of the table columns
3. Add a data-link for that column
4. For the link-title, enter the text `View Cardinality Overview dashboard scoped to ${label}: ${__data.fields.value}`. The URL can be left blank
5. Now see that the edit ✏️ and delete ❌  icons quite easily disappear off the screen when the panel-edit side-panel isn't large enough.

**What you expected to happen**:
The edit and delete buttons should always be on the screen, no matter how wide or narrow I set the panel-edit side-panel, and no matter what content has been set in that specific data-link.

**How to reproduce it (as minimally and precisely as possible)**:

1. Sign up for GrafanaCloud free account
6. Go to Cardinality Management 1: overview
7. Make it editable
8. Edit "Highest cardinality metrics" panel
9. Try to edit the data-link, on a narrow panel-edit side-panel.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.4.0-42834pre
- Data source type & version: Prometheus - `grafanacloud-usage` datasource on GrafanaCloud
- OS Grafana is installed on: GrafanaCloud
- User OS & Browser: macOS 12.6.1 (21G217), Version 107.0.5304.110 (Official Build) (arm64)
Automatically update Chrome for all users [Learn more](https://support.google.com/chrome/answer/95414)

### PR DESCRIPTION
**Pull request by dprokop**
Fixes https://github.com/grafana/grafana/issues/59150

### CODE REVIEW

File: packages/grafana-ui/src/components/DataLinks/DataLinksInlineEditor/DataLinksListItem.tsx [hunk 1]
```diff
@@ -63,6 +60,19 @@ const getDataLinkListItemStyles = stylesFactory((theme: GrafanaTheme2) => {
...
+    `,
+    titleWrapper: css`
+      label: data-links-list-item-title;
+      justify-content: space-between;
+      display: flex;
+      width: 100%;
+      align-items: center;
+    `,
+    actionButtons: css`
+      margin-left: 8px;
```
Comment (ivanortegaalba) [inline packages/grafana-ui/src/components/DataLinks/DataLinksInlineEditor/DataLinksListItem.tsx:74]: Don't we have any theme variable for this?

**Review by ivanortegaalba — APPROVED** [review]
Looks great to me! LGTM!

File: packages/grafana-ui/src/components/DataLinks/DataLinksInlineEditor/DataLinksListItem.tsx (hunk 1 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/DataLinks/DataLinksInlineEditor/DataLinksListItem.tsx:74]: yes, need to updaye

**Comment by ivanortegaalba** [comment]
Thanks for applying feedback, LGTM

**Pull request merged**